### PR TITLE
Improved Hadoop Collector

### PR DESCRIPTION
**Pull request by aaronabf**
Diamond has custom Hadoop and HBase collectors. Both are nearly identical and face the same problems:

1. Written for Hadoop Metrics, not Hadoop Metrics2, the latter of which we use
2. Improper/poor handling of paths, hosts, and timestamps
3. Makes a direct call to `publish_metric` which ignores the metric white/blacklists

This revision fixes these issues, cleans up the code, and removes the irrelevant HBase collector.

Note: I still need to fix the tests.

### CODE REVIEW

File: src/collectors/hbase/hbase.py [hunk 1]
```diff
@@ -1,110 +0,0 @@
-# coding=utf-8
-
-"""
-Diamond collector for HBase metrics, see:
```
Comment (gregsterin) [inline src/collectors/hbase/hbase.py]: Why delete this? If we don't need it lets just not configure it as a collector?

File: src/collectors/hadoop/test/testhadoop.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class TestHadoopCollector(CollectorTestCase):
 
     def setUp(self):
         config = get_collector_config('HadoopCollector', {
-            'metrics':  [os.path.dirname(__file__) + '/fixtures/*metrics.log'],
+            'metrics_files':  [os.path.dirname(__file__) + '/fixtures/*metrics.log'],
```
Comment (gregsterin) [inline src/collectors/hadoop/test/testhadoop.py]: dont see the new fixtures files though for the new format.  git add?

Comment (aaronabf) [reply src/collectors/hadoop/test/testhadoop.py]: They were being git ignored, removed the ignore.

File: src/collectors/hbase/hbase.py (hunk 1 above)
Comment (aaronabf) [reply src/collectors/hbase/hbase.py]: There is no reason anyone would ever want to use it; It's buggy and lacking functionality. Removing it improves the quality of the code by preventing anyone from mistakenly using it.

Comment (mahendra) [reply src/collectors/hbase/hbase.py]: @aaronabf as an opensource project, Diamond may have users of the older Hadoop Metrics Infra. I would prefer that we do not break compatibility.

We should ideally implement a new collector and leave the old one there - can be deprecated if the upstream maintainers agree on it.

**Comment by mahendra** [comment]
@EvanKuhn-Affirm fyi - would be good for you to take a look as well

File: src/collectors/hbase/hbase.py (hunk 1 above)
Comment (aaronabf) [reply src/collectors/hbase/hbase.py]: I would normally agree, but these collectors are straight-up buggy: they ignore documented configuration options and provide misleading timestamp info. Regardless, I wouldn't want to add this back. I would be fine having two collectors, for each metrics version, but these current collectors should be formally deprecated. We can have: `HadoopCollectorMetrics1` and `HadoopCollectorMetrics2`.

**Comment by aaronabf** [comment]
Moved code to a new collector `HadoopMetrics2Collector`. All tests passing.

File: src/collectors/hadoop2/hadoop.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            host = extra_data.get('Hostname', None) or self.get_hostname()
+            partial_path = 'hadoop.{}.'.format(raw_data['name'])
+
+            for key, val in metrics.items():
+                full_path = partial_path + key
+                self._publish(key, val, full_path, host, raw_data['timestamp'])
+
+        if str_to_bool(self.config['truncate']):
+            _file.seek(0)
+            _file.truncate()
```
Comment (gregsterin) [inline src/collectors/hadoop2/hadoop.py]: One concern about this. What happens if data gets written between when we reach current EOF and when we go to truncate the file?

We would accidentally delete lines we havent drained yet? Or does the file handle only truncate what we have read. Can you do a quick test to verify... if we have to worry about this, we can track the size of the lines as we read them, and then truncate from start with the size parameter to truncate(), which should work.

**Review by gregsterin — COMMENTED** [review]
lgtm, one comment about truncation that was already there before but seems suspect.

File: src/collectors/hadoop2/hadoop.py (hunk 3 above)
Comment (aaronabf) [reply src/collectors/hadoop2/hadoop.py]: I modified the code to first get the file size as soon as the file is open:
```
_file = open(filename, 'r')
original_file_end = _get_file_end(_file)
```
then when we're finished reporting metrics, truncate the file up to this size:
```
with open(filename, 'r+') as _file:
    new_file_end = _get_file_end(_file)
     _file.truncate(new_file_end - original_file_end)
```

**Review by gregsterin — APPROVED** [review]
lgtm. Perhaps we should add a test for the truncate-only-what-weve-read logic?

**Comment by aaronabf** [comment]
Merging in for now. I will make a note to look into Hadoop to see if they handle moved files.

**Pull request merged**